Library/Homebrew/dependency_collector.rb

```diff
@@ -30,12 +30,18 @@ def initialize
     @requirements = Requirements.new
   end
 
-  def initialize_copy(other)
+  def initialize_dup(other)
     super
     @deps = @deps.dup
     @requirements = @requirements.dup
   end
 
+  def freeze
+    @deps.freeze
+    @requirements.freeze
```

Comment:
In all these cases: would be good to add a comment in the constructor pointing out that any instance variables like this should be frozen/duped.